service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -449,4 +455,35 @@ public Map<String, String> pipelineRunOutputsAsMap(String outputsString) {
       throw new InternalServerErrorException("Error reading pipeline run outputs", e);
     }
   }
+
+  /**
+   * Extract a paginated list of Pipeline Run records from the database
+   *
+   * @param pageSpecification - defines what field to paginate over and what page to start from for
+   *     the query
+   * @param userIdSpecification - defines what field and value to filter on pre pagination i.e.
+   *     filter on only records for user x
+   * @param pageable - defines how many records to return and is used to encode the results previous
+   *     and next page token
+   * @return - a PageResponse containing the list of records in the current page and the page tokens
+   *     for the next and previous page if applicable
+   */
+  public PageResponse<List<PipelineRun>> findPageResults(
```

Comment:
could this method name be more descriptive? like `findPipelineRunsPaginated` or `findPipelineRunPageResults`